keras/engine/topology.py

```diff
@@ -2867,10 +2868,72 @@ def _collect_input_shape(input_tensors):
     return shapes
 
 
+def _save_attributes_to_hdf5_group(group, name, data):
+    """Saves attributes (data) of the specified name into the HDF5 group.
+    This method deals with an inherent problem of HDF5 file which is not
+    able to store data larger than HDF5_OBJECT_HEADER_LIMIT bytes.
+
+    # Arguments
+        group: A pointer to a HDF5 group.
+        name: A name of the attributes to save.
+        data: Attributes data to store.
+    """
+    # Check that no item in `data` is larger than `HDF5_OBJECT_HEADER_LIMIT`
+    # because in that case even chunking the array would not make the saving
+    # possible.
+    bad_attributes = [x for x in data if len(x) > HDF5_OBJECT_HEADER_LIMIT]
+
+    # Expecting this to never be true.
+    if len(bad_attributes) > 0:
+        raise RuntimeError("the following attributes cannot be saved to HDF5 file "
+                           "because they are larger than %d bytes: '%s'"
+                           % (HDF5_OBJECT_HEADER_LIMIT,
+                              "', '".join([x for x in bad_attributes])))
+
+    data_npy = np.asarray(data)
```

Comment:
Don't we already have a guarantee that `data` is a numpy array?

Reply:
perhaps yes as it is used now, but this choice seems reasonable to me. I can remove it if desired I don't feel strongly.